From 9347bcf55223d3bc04ed8f1b6aa8a6d75bf66f09 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 26 Sep 2022 14:16:37 +0300
Subject: [PATCH] [SW-0] gaudi3: remove validation pass check for erom

For erom images, ROM always perform boot desc init, dma and load
erom images. Since boot_desc_init (dma of manifest image) is always
done for erom images, signature verification of manifest should be
always done (img_validation=1) for erom manifest and should not
be dependent on validation pass flag. This patch add changes to remove
validation pass changes for erom manifest.

Change-Id: I1d86d34e2107188dd3390548e0de3ac8bd80b06d
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 77357fe..573ec0a 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -973,7 +973,10 @@
 	uint32_t img_plus_key_len;
 	uint8_t crc_bytes, num_chunks;
 
-	if ((arcboot_rom_global.validation_pass) || (!security_efuse_image_validation_en()))
+	if (!security_efuse_image_validation_en())
+		return 0;
+
+	if ((img == HL_FW_BOOT_PTR_SPI) && (arcboot_rom_global.validation_pass))
 		return 0;
 
 	if (img == HL_FW_BOOT_PTR_EROM)
